### PR TITLE
Fix filter not resetting on bookmarks

### LINKED ISSUE #15156: Cannot remove filter in preset
### Describe the Bug

After setting a filter in a preset, it is impossible to clear it completely

### To Reproduce

1. Create a preset on a collection and save it
2. Add a filter and save it
3. Reopen the preset and try to clear the filter, either from the drop-down or clicking the` x`

### Errors Shown

You lay get ab error about Discard changes or Continue editing
Anyway, the filter is still there 

### What version of Directus are you using?

9.15.1

### What version of Node.js are you using?

16.14.0

### What database are you using?

PostgreSQL

### What browser are you using?

Firefox

### How are you deploying Directus?

locally

### PR DESCRIPTION
**Pull request by Nitwel**
closes #15156 

improves upon #14791

### CODE REVIEW

File: app/src/modules/settings/routes/presets/item.vue [hunk 1]
```diff
@@ -153,446 +153,423 @@ type FormattedPreset = {
...
-						options: {
-							placeholder: t('preset_name_placeholder'),
-						},
-					},
+function useLinks() {
+	const backLink = computed(() => {
+		return `/settings/presets`;
+	});
+
+	return { backLink };
```
Comment (br41nslug) [inline app/src/modules/settings/routes/presets/item.vue:440]: making this into a computed seems like overkill to me as it's never changed by the looks of it 🤔 or im missing something

Comment (br41nslug) [reply app/src/modules/settings/routes/presets/item.vue:440]: But then again it already existed and was just moved i see 😕

File: app/src/modules/settings/routes/presets/item.vue [hunk 2]
```diff
@@ -129,9 +129,9 @@
 	</component>
 </template>
 
-<script lang="ts">
+<script setup lang="ts">
```
Comment (br41nslug) [inline app/src/modules/settings/routes/presets/item.vue:132]: Just out of curiosity, what are the benefits/arguments for using this `<script setup` form over the `defineComponent` form? the latter feels more declarative to me but as a relative newcomer to Vue having all the different forms can feel a bit like a mess in the ecosystem sometimes 😓

**Review by br41nslug — APPROVED** [review]
Looks like a much more robust solution than my original one 😄 Tested both the new and old issue and solves them nicely

File: app/src/modules/settings/routes/presets/item.vue (hunk 1 above)
Comment (Nitwel) [reply app/src/modules/settings/routes/presets/item.vue:440]: I'll just clean that up even though it was already there before.

File: app/src/modules/settings/routes/presets/item.vue (hunk 2 above)
Comment (Nitwel) [reply app/src/modules/settings/routes/presets/item.vue:132]: `script setup` is funnily more declarative as with it you define props and emits way more intuitive and the main benefit of using script setup is way less overhead. When you write components using `defineComponent` you have to mannually expose each variable and function to the template which comes annoying really quickly where as on `script setup`, everything is public by default.

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**